src/qr/qrcodegen.nim

```diff
@@ -54,19 +54,10 @@
 
 import os
 
+const libname* {.strdefine.} = currentSourcePath().parentDir / "qrcodegen." & (when defined(windows): "dll" elif defined(macosx): "dylib" else: "so") ## Dynamic Library
```

Comment:
I can't get `{.strdefine.}` to work. When compiling with `-d:libname=/usr/lib/qrcodegen.so` the executable still uses the nimble path.

Reply:
Try with compiling again with `-f` ?, maybe add quotes `" "` to the path ?, weird it works on my machine. :thinking:

Reply:
Well, this is weird.. I tried with quotes and all.. I have looked into the cache directory. See this:

`@ma.nim.c`
```
/* section: NIM_merge_DATA */
STRING_LITERAL(TM__LSU9bViOd4immx1fl6MvEHQ_3, "/usr/lib/qrcodegen.so", 21);
static NIM_CONST tyArray__nHXaesL0DJZHyVS07ARPRA TM__LSU9bViOd4immx1fl6MvEHQ_2 = {((NimStringDesc*) &TM__LSU9bViOd4immx1fl6MvEHQ_3)}
;
```

`@msrc@sqr.nim.c`
```
/* section: NIM_merge_DATA */
STRING_LITERAL(TM__DW1kcEEkG4t559ckyjNU1Cw_4, "/home/user/git/nim_qr/src/qr/qrcodegen.so", 41);
STRING_LITERAL(TM__DW1kcEEkG4t559ckyjNU1Cw_5, "/home/user/git/nim_qr/src/qr/qrcodegen.so", 41);
static NIM_CONST tyArray__HU7qaqKu9czJLT84iCBJnsA TM__DW1kcEEkG4t559ckyjNU1Cw_10 = {((NI) 4),
((NI) 7)}
;
```

Reply:
Use an empty folder as cache, like `-f --nimcache:.` or `-f --nimcache:/tmp` ?. :thinking:

Reply:
Yeah - this is strange. I have also used the `--nimcache:` option, but with no luck.

Could you check your generated C-files?

`@m..@s..@s.nimble@spkgs@sqr-1.5.0@sqr.nim.c`
```
/* section: NIM_merge_DATA */
STRING_LITERAL(TM__DW1kcEEkG4t559ckyjNU1Cw_4, "/home/user/.nimble/pkgs/qr-1.5.0/qr/qrcodegen.so", 48);
STRING_LITERAL(TM__DW1kcEEkG4t559ckyjNU1Cw_5, "/home/user/.nimble/pkgs/qr-1.5.0/qr/qrcodegen.so", 48);
```

Reply:
It seems to be working on a simple example:

```console
$ cat a.nim 
const x* {.strdefine.} = "ABCDEF0123456"
echo x

$ nim c -r -d:danger -d:x:hola --nimcache:. a.nim 

$ cat @ma.nim.c | grep "ABCDEF0123456"

$ cat @ma.nim.c | grep "hola"
STRING_LITERAL(TM__R8RUzYq41iOx0I9bZH5Nyrw_3, "hola", 4);

$ 
```

Reply:
So it seems `{.strdefine.}` may not be reliable for this use case, the ones on stdlib do not use it,
do you want me to remove `{.strdefine.}` and go with the `try: ... except:` ?.

Reply:
I can also make the small example work.. well, another solution it is. A `try: ... except:` would be fine with me.

1) I would prefer, that we start by looking in the default library folder, `/usr/lib/qrcodegen.so` (don't know with Windows and Mac), if it does not exists here, then we'll use the nimble. How does that sound for portability?

2) If we go with nr. 1, the `qr.nimble` should check in that folder before compiling it from C?

Reply:
@ThomasTJdev Something like this ?.
https://github.com/ThomasTJdev/nim_qr/pull/5/commits/2e54ca37f8cd01327061555d87b1ecb5f730150f

I dont know if this would work 100% of times, because needs sudo, Windows may block it, file permissions, antivirus, etc
but ultimately it prints clear instructions about copying the file manually.

Reply:
Yeah - I like that solution. It wont work out of the box due to the need of `sudo`, and we shouldn't run `nimble install` with `sudo`. The above works perfect for me cause I have the file in `/usr/lib`.

But. What if the user wants to ship the nimble-package with `qrcodegen.so`? Well, both the previous design and this PR does not give the user that possibility due to the `lib` being an absolute path. So we are blocking the *no-interaction* portability. To fix that it should be possible to use a relative path to the lib, or the lib should be included in the compiled file.

How does this sound to you:
* Merging this PR and updating the `README.md` with a clear description of it.
* Await another user with an innovative idea on a solution :)

Reply:
OK with the Merge. :+1: 

I think theres no silver bullet solution, shared libs are meant to work like that.